### PR TITLE
Fix #3 and #4, new integration tests

### LINKED ISSUE #3: Furnace creates a css file when only core is selected

### PR DESCRIPTION
**Pull request by alex-page**
There was an issue with core only and no javascript builds.

The integration tests had to be completely rewritten as they relied on modules which caused them to be updated all the time.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # :fire: Furnace
 
-> Melts down the [GOLD](https://github.com/govau/gold) components into a zip file.
+> Melts down the [GOLD](https://github.com/govau/designsystem) components into a zip file.
```
Comment (dominikwilkowski) [inline README.md]: You also wanna change the text in the link from `GOLD` to `design system`

File: __tests__/integration/mocks/node_modules/@gov.au/_test-00/lib/sass/_globals.scss [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*! @gov.au/core v1.0.0 */
+/*! @gov.au/_test-00 v1.0.1 */
```
Comment (dominikwilkowski) [inline __tests__/integration/mocks/node_modules/@gov.au/_test-00/lib/sass/_globals.scss]: Why is the core included in git? I see this is the `_test-00` mock file but it shouldn't have any sass in it no?

File: __tests__/integration/test-03/fixture/AU-DesignSystem/node_modules/@gov.au/_test-00/lib/sass/_globals.scss [hunk 3]
```diff
@@ -0,0 +1,798 @@
+/*! @gov.au/_test-00 v1.0.1 */
```
Comment (dominikwilkowski) [inline __tests__/integration/test-03/fixture/AU-DesignSystem/node_modules/@gov.au/_test-00/lib/sass/_globals.scss:1]: That's a bit much for a test no?

File: uikit.json [hunk 4]
```diff
@@ -1 +1 @@
...
...path":"lib/js/react.js"}}},"@gov.au/skip-link":{"name":"@gov.au/skip-link","version":"1.0.0","dependencies":{"@gov.au/core":">=1.0.0"},"path":"/Users/alexpage/Sites/dta/furnace/node_modules/@gov.au/skip-link","settings":{"version":"1.0.0","plugins":["@gov.au/pancake-sass","@gov.au/pancake-react","@gov.au/pancake-json"],"sass":{"path":"lib/sass/_module.scss","sass-versioning":true},"react":{"path":"lib/js/react.js"}}},"@gov.au/tags":{"name":"@gov.au/tags","version":"1.0.0","dependencies":{"@gov.au/core":">=1.0.0"},"path":"/Users/alexpage/Sites/dta/furnace/node_modules/@gov.au/tags","settings":{"version":"1.0.0","plugins":["@gov.au/pancake-sass","@gov.au/pancake-react","@gov.au/pancake-json"],"sass":{"path":"lib/sass/_module.scss","sass-versioning":true},"react":{"path":"lib/js/react.js"}}},"@gov.au/text-inputs":{"name":"@gov.au/text-inputs","version":"1.0.0","dependencies":{"@gov.au/core":">=1.0.0"},"path":"/Users/alexpage/Sites/dta/furnace/node_modules/@gov.au/text-inputs","settings":{"version":"1.0.0","plugins":["@gov.au/pancake-sass","@gov.au/pancake-react","@gov.au/pancake-json"],"sass":{"path":"lib/sass/_module.scss","sass-versioning":true},"react":{"path":"lib/js/react.js"}}}}
```
Comment (dominikwilkowski) [inline uikit.json]: This file is generated by pancake-json right? Should be ignored from git

File: package.json [hunk 5]
```diff
@@ -26,11 +26,11 @@
 		"unit-test:watch": "jest --watch --coverage",
 		"integration-test": "node __tests__/integration/tester.js",
 		"integration-test:verbose": "node __tests__/integration/tester.js --verbose",
-		"integration-test:watch": "nodemon --watch __tests__/integration --exec npm run integration-test:verbose",
-		"start": "node dist/init.js",
+		"integration-test:start": "node dist/index.js --verbose -j ../__tests__/integration/mocks/uikit.json",
```
Comment (dominikwilkowski) [inline package.json]: Did we not have a test runner? That runs multiple tests?

**Review by dominikwilkowski — CHANGES_REQUESTED** [review]

File: __tests__/integration/mocks/node_modules/@gov.au/_test-00/lib/sass/_globals.scss (hunk 2 above)
Comment (alex-page) [reply __tests__/integration/mocks/node_modules/@gov.au/_test-00/lib/sass/_globals.scss]: This is a test for a module with sass that doesn't compile to any css. Maybe this is something that should be fixed.

File: uikit.json (hunk 4 above)
Comment (alex-page) [reply uikit.json]: Definitely should be ignored

File: README.md (hunk 1 above)
Comment (alex-page) [reply README.md]: Yeah I'll do that

File: __tests__/integration/test-03/fixture/AU-DesignSystem/node_modules/@gov.au/_test-00/lib/sass/_globals.scss (hunk 3 above)
Comment (alex-page) [reply __tests__/integration/test-03/fixture/AU-DesignSystem/node_modules/@gov.au/_test-00/lib/sass/_globals.scss:1]: Yeah I have left only the comment now

File: package.json (hunk 5 above)
Comment (alex-page) [reply package.json]: That happens here in code:
https://github.com/govau/furnace/blob/develop/__tests__/integration/tester.js#L151

Comment (dominikwilkowski) [reply package.json]: I think we need to talk about this one again :)

Comment (alex-page) [reply package.json]: I have removed the double test run from inside the tester file as we actually run 6 tests now

File: .gitignore [hunk 6]
```diff
@@ -3,23 +3,30 @@
...
 npm-debug.log
 lerna-debug.log
+uikit.json
 
 # Jest files
 dist
 coverage
 
-# Don’t commit temp test files
+# Don't commit temp test files
```
Comment (dominikwilkowski) [inline .gitignore]: seriously!

File: README.md [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # :fire: Furnace
 
-> Melts down the [GOLD](https://github.com/govau/gold) components into a zip file.
+> Melts down the [GOLD ( Design System )](https://github.com/govau/designsystem) components into a zip file.
```
Comment (dominikwilkowski) [inline README.md:3]: remove the word `GOLD` and replace with `Design System` please :(

**Review by dominikwilkowski — APPROVED** [review]

File: README.md (hunk 7 above)
Comment (alex-page) [reply README.md:3]: 😬 no

**Pull request merged**